src/include/buffer.h

```diff
@@ -957,7 +956,7 @@ namespace buffer CEPH_BUFFER_API {
     }
     uint32_t crc32c(uint32_t crc) const;
     void invalidate_crc();
-    boost::optional<sha1_digest_t> sha1(); 
+    std::pair<bool, sha1_digest_t> sha1(); 
```

Comment:
might want to `#include <utility>` explicitly, instead of relying on other included headers to do so.